src/Microsoft.Diagnostics.DebugServices.Implementation/RuntimeProvider.cs

```diff
@@ -28,6 +28,10 @@ public RuntimeProvider(IServiceProvider services)
         /// <param name="startingRuntimeId">The starting runtime id for this provider</param>
         public IEnumerable<IRuntime> EnumerateRuntimes(int startingRuntimeId)
         {
+            // The DataTarget isn't cached here because there is no way to flush it. It currently caches
+            // ClrInfos, modules and PE images. It also isn't disposed because the DataTarget instance
+            // needs to be around as long as the ClrInfos wrapped by the Runtime instances created below
+            // which the life time is determined by the RuntimeService.
```

Comment:
Does this mean that we should try moving this to the runtime service? Just to track lifetimes closer and not have these bugs.

Reply:
There really isn't any bug here. This is just a reminder of what is going on.  The DataTarget/CustomDataTarget instances are just never disposed which is fine since a new one is created when the Runtime instances are created on Flush.  

This code can't be moved to the RuntimeService because it doesn't know anything about create CLRMD runtimes the whole point of separating the RuntimeProviders from the RuntimeService.